numpy/core/_objectmath.py

```diff
@@ -0,0 +1,262 @@
+"""
+from __future__ import division, absolute_import, print_function
+import math
+from numpy.core.umath import pi as PI
+import numpy as np
+import sys
+
+LOG2 = math.log(2)
+
+if sys.version_info > (3,):
+    def isint(x):
+        return type(x) in (bool, int)
+else:
+    def isint(x):
+        return type(x) in (bool, int, long)
+
+def float_or_complex(x):
+    return type(x) in (float, complex)
+
+def _notimplemented_msg(fname, etype, etype2=None):
+    if etype2 is not None:
+        typestr = "({a}, {b})".format(a=etype, b=etype2)
+    else:
+        typestr = str(etype)
+    return ("object-array fallback for ufunc '{name}' is not implemented for "
+            "arguments of type {type}").format(name=fname, type=typestr)
+
+
+def logical_and(x, y):
+    return bool(x and y)
+
+def logical_or(x, y):
+    return bool(x or y)
+
+def logical_xor(x, y):
+    return bool(x or y) and not bool(x and y)
+
+def logical_not(x):
+    return bool(not x)
+
+def maximum(x, y):
+    # get nan and complex right when casting
+    if isint(y) and float_or_complex(x):
+        return np.maximum(x, float(y)).item()
+    if isint(x) and float_or_complex(y):
+        return np.maximum(float(x), y).item()
+    return max(x, y)
+
+def minimum(x, y):
+    if isint(y) and float_or_complex(x):
+        return np.minimum(x, float(y)).item()
+    if isint(x) and float_or_complex(y):
+        return np.minimum(float(x), y).item()
+    return min(x, y)
+
+def fmax(x, y):
+    if isint(y) and float_or_complex(x):
+        return np.fmax(x, float(y)).item()
+    if isint(x) and float_or_complex(y):
+        return np.fmax(float(x), y).item()
+    return max(x, y)
+
+def fmin(x, y):
+    if isint(y) and float_or_complex(x):
+        return np.fmin(x, float(y)).item()
+    if isint(x) and float_or_complex(y):
+        return np.fmin(float(x), y).item()
+    return min(x, y)
+
+def iscomplex(x): # not actually a ufunc (yet)
+    if isint(x):
+        return False
+    raise TypeError(_notimplemented_msg('iscomplex', type(x)))
+
+def isreal(x): # not actually a ufunc (yet)
+    if isint(x):
+        return True
+    raise TypeError(_notimplemented_msg('isreal', type(x)))
+
+def isnan(x):
+    if isint(x):
+        return False
+    raise TypeError(_notimplemented_msg('isnan', type(x)))
+
+def isinf(x):
+    if isint(x):
+        return False
+    raise TypeError(_notimplemented_msg('isinf', type(x)))
+
+def isfinite(x):
+    if isint(x):
+        return True
+    raise TypeError(_notimplemented_msg('isfinite', type(x)))
+
+def conjugate(x):
+    if isint(x):
+        return x
+    raise TypeError(_notimplemented_msg('conjugate', type(x)))
+
+def degrees(x):
+    return x*180/PI
+rad2deg = degrees
+
+def radians(x):
+    return x*PI/180
+deg2rad = radians
+
+def square(x):
+    return x*x
+
+def reciprocal(x):
+    if isint(x):
+        if x == 1:
+            return 1
+        if x == -1:
+            return -1
+        if x == 0:
+            return np.reciprocal(0).item()
```

Comment:
This gives `RuntimeWarning: divide by zero encountered in reciprocal` - was that intended?

Reply:
I don't remember the intention anymore, but maybe. Doesn't the corresponding int32/int64 ufunc raise the warning too? Then the object ufunc probably should as well.

```python
np.reciprocal([1,0,1])                                                    
/usr/bin/ipython3:1: RuntimeWarning: divide by zero encountered in reciprocal
  #!/usr/bin/python3
/usr/bin/ipython3:1: RuntimeWarning: invalid value encountered in reciprocal
  #!/usr/bin/python3
array([                   1, -9223372036854775808,                    1])
```